core/new-gui/src/app/workspace/service/workflow-graph/model/workflow-action.service.ts

```diff
@@ -1153,4 +1280,21 @@ export class WorkflowActionService {
       this.getJointGraphWrapper().showLinkBreakpoint(linkID);
     }
   }
+
+  private sendCommand(update: string): void {
+    if (this.workflowCollabService.getSendData()) {
+      this.workflowCollabService.sendCommand(update);
+    }
+  }
+
+  private handleRemoteChange(): void {
+    this.workflowCollabService.getCommandMessageStream().subscribe(message => {
+      if (message.type === "execute") {
+        this.toggleSendData(false);
+        const func = message.action;
+        (this[func] as any).apply(this, message.parameters);
+        this.toggleSendData(true);
+      }
+    });
+  }
```

Comment:
I think I've seen this code somewhere else. it seems to be a duplicate. please get rid of duplicates as much as possible as they are not easy to maintain in the future.

Reply:
`undo-redo.service.ts` also has this method, but with different content, so this is not duplicate.

Reply:
they look similar enough. can you try unify them? 

also, please let me resolve the comment if it is raised by me. otherwise it's easily to lose track of progress.

Reply:
Done.